cmd/deploy/deploy.go

```diff
@@ -286,18 +287,28 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 		return oktetoErrors.ErrDeployCantDeploySvcsIfNotCompose
 	}
 
+	// We need to create a client that doesn't go through the proxy to create
+	// the configmap without the deployedByLabel
+	c, _, err := dc.K8sClientProvider.Provide(okteto.Context().Cfg)
+	if err != nil {
+		return err
+	}
+
 	cwd, err := os.Getwd()
 	if err != nil {
 		return fmt.Errorf("failed to get the current working directory: %w", err)
 	}
 
-	// We need to create a client that doesn't go through the proxy to create
-	// the configmap without the deployedByLabel
-	c, _, err := dc.K8sClientProvider.Provide(okteto.Context().Cfg)
+	topLevelGitDir, err := repository.FindTopLevelGitDir(cwd, dc.Fs)
 	if err != nil {
-		return err
+		oktetoLog.Warning("Repository not detected: the env vars 'OKTETO_GIT_BRANCH' and 'OKTETO_GIT_COMMIT' might not be available: %s.\n    For more information, check out: https://www.okteto.com/docs/cloud/okteto-cli/#built-in-environment-variables", err.Error())
```

Comment:
can we use the values of the constant for message formatting?

Reply:
good idea, changed in https://github.com/okteto/okteto/pull/4080/commits/50b0fc0cd2b043329f6497ec5776872c741c8ea9#diff-15a334b420072b3d7ab3589a233e57b32c74c7b83ec2af01e0c939cf3ef11afa